rosbag2_storage_mcap/src/mcap_storage.cpp

```diff
@@ -841,12 +841,245 @@ void MCAPStorage::ensure_rosdistro_metadata_added()
   has_added_ros_distro_metadata_ = true;
 }
 
+/**
+ * A storage implementation for the MCAP file format of snapshots.
+ */
+class ROSBAG2_STORAGE_MCAP_PUBLIC MCAPStorageSnapshot
+: public rosbag2_storage::storage_interfaces::ReadWriteInterface
+{
+public:
+  MCAPStorageSnapshot();
+  ~MCAPStorageSnapshot() override;
+  /** BaseIOInterface **/
+  #ifdef ROSBAG2_STORAGE_MCAP_HAS_STORAGE_OPTIONS
+    void open(const rosbag2_storage::StorageOptions & storage_options,
+              rosbag2_storage::storage_interfaces::IOFlag io_flag =
+                rosbag2_storage::storage_interfaces::IOFlag::READ_WRITE) override;
+    void open(const std::string & uri, rosbag2_storage::storage_interfaces::IOFlag io_flag =
+                                         rosbag2_storage::storage_interfaces::IOFlag::READ_WRITE);
+  #else
+    void open(const std::string & uri,
+            rosbag2_storage::storage_interfaces::IOFlag io_flag =
+            rosbag2_storage::storage_interfaces::IOFlag::READ_WRITE) override;
+  #endif
+
+
+  /** BaseInfoInterface **/
+  rosbag2_storage::BagMetadata get_metadata() override;
+  std::string get_relative_file_path() const override;
```

Comment:
I assume the original function did the same thing? So why override it? Just leave it out I think.